### ISSUE TITLE
how to stateless ignore "react/prop-types"?

### ISSUE THREAD

**Issue by LiaoSept** [yannickcr/eslint-plugin-react#480]

**Comment by ljharb** [comment]
@LiaoSept you don't, nor would you want to. Here's an example:
```js
function Foo(props) {
  return <div className={props.className} />
}
Foo.propTypes = {
  className: React.PropTypes.string.isRequired
};
```

Obviously you can use `/* eslint react/prop-types: 0 */` but that's not a good idea.

**Comment by LiaoSept** [comment]
@ljharb I just want to ignore the stateless

**Comment by ljharb** [comment]
@LiaoSept why? stateless components need propTypes exactly the same as non-stateless components.

**Comment by LiaoSept** [comment]
internal use, not exposed to the outside

**Comment by ljharb** [comment]
That's not true of all stateless components tho - if you want to ignore propTypes on internal modules (which imo you shouldn't) then you'd use an eslint comment in those files.

**Comment by LiaoSept** [comment]
example:
```jsx
const ItemList = React.createClass({
    render() {
         return (
            <div clsssName="item-list">
                {_.map(this.props.items, user => <Item item={item}/>)}
            </div>
        );
    }
});

const Item = ({item}) => (
    <div className="item">{item}</div>
);
```

**Comment by ljharb** [comment]
so you should add `Item.propTypes = { item: React.PropTypes.node };`
